.travis.yml

```diff
@@ -1,9 +1,34 @@
-dist: trusty
-sudo: true
+stages:
+  - name: lint
+  - name: test
+  - name: integration
+    if: type = cron OR commit_message =~ /\[ci integration\]/ OR (tag IS present AND tag =~ /^v[\d]+\.[\d]+\.[\d]+/)
+  - name: deploy
+    if: tag IS present AND tag =~ /^v[\d]+\.[\d]+\.[\d]+/
+
+templates:
+  job-template-test: &job-template-test
+    before_install:
+      - mkdir -p $HOME/.bin
+      - export PATH=$PATH:$HOME/.bin
+      - .travis/before_install.sh
+    install:
+      - .travis/install.sh
+    script:
+      - .travis/run.sh
+    after_script:
+      - coveralls
+
+  job-template-integration: &job-template-integration
```

Comment:
question: So a job template is just all the steps you have in the template. And this particular one is just like the `job-template-test` but with sudo enabled full VMs

Reply:
Yes, this a taking advantage of [YAML anchors](https://blog.daemonl.com/2016/02/yaml.html).
It's just expanding the 'template' into the job definitions and in this case adding the `sudo` setting.